src/SprykerSdk/Spryk/Model/Spryk/Dumper/SprykDefinitionDumper.php

```diff
@@ -33,16 +33,20 @@ public function __construct(SprykDefinitionFinderInterface $definitionFinder, Sp
     }
 
     /**
+     * @param int|null $level
+     *
      * @return array
      */
-    public function dump(): array
+    public function dump(?int $level = null): array
     {
         $sprykDefinitions = [];
         foreach ($this->definitionFinder->find() as $fileInfo) {
             $sprykName = str_replace('.' . $fileInfo->getExtension(), '', $fileInfo->getFilename());
             $sprykDefinition = $this->configurationLoader->loadSpryk($sprykName);
 
-            $sprykDefinitions[$sprykName] = $sprykDefinition;
+            if ($level === null || $level === (int) $sprykDefinition['level']) {
```

Comment:
It looks like a validation logic, could be moved to a separate method. But it's ok to leave it like this. Also please note that we're using the casting functions like `(int)` wihtout whitespaces: `(int)$smth`.